File: Programmers/Lv2/모음사전.py
import itertools

def solution(word):
    answer = 0
    words = ['A','E','I','O','U']
    result = set()
    for i in range(1, 6):
        tmp = list(itertools.product(words, repeat=i))
        for w in tmp:
            k = ''
            for j in range(len(w)):
                k += w[j]
            result.add(k)
    res = list(result)
    res.sort()
    cnt = 0
    for i in range(len(res)):
        cnt += 1
        if res[i] == word:
            return cnt

File: Programmers/Lv2/모음 사전.js
function solution(word) {
  var vowel = ['A', 'E', 'I', 'O', 'U'];
  var tmp = new Set();

  const dfs = (out) => {
    if (out.length > 5) {
      return;
    }
    tmp.add(out);
    for (let i = 0; i < 5; i++) {
      dfs(out + vowel[i]);
    }
  };

  dfs('');
  var cvt = Array.from(tmp).sort();
  // var cvt = Array(...tmp).sort();

  for (let i = 0; i < cvt.length; i++) {
    if (cvt[i] === word) {
      return i;
    }
  }
}
